sky/packages/sky/lib/rendering/object.dart

```diff
@@ -485,6 +590,28 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
     _needsLayout = false;
     markNeedsPaint();
   }
+
+  /// Compute the layout for this render object
+  ///
+  /// This function is the main entry point for parents to ask their children to
+  /// update their layout information. The parent passes a constraints object,
+  /// which informs the child as which layouts are permissible. The child is
+  /// required to obey the given constraints.
+  ///
+  /// If the parent reads information computed during the child's layout, the
+  /// parent must pass true for parentUsesSize. In that case, the parent will be
+  /// marked as needing layout whenever the child is marked as needing layout
+  /// because the parent's layout information depends on the child's layout
+  /// information. If the parent passes false for parentUsesSize, the child can
+  /// change its layout information (subject to the given constraints) without
+  /// informing the parent.
+  ///
+  /// Subclasses should not override layout directly. Instead, they should
+  /// override performResize and/or performLayout.
+  ///
+  /// Parent should call layout unconditionally. It is layout's responsibility
```

Comment:
Make that: The parent's performLayout method should call the layout of all its children unconditionally.